File: app/src/main/java/com/futuretech/nfmovies/Utils/APIUtil.java
package com.futuretech.nfmovies.Utils;

import com.futuretech.nfmovies.API.*;

public class APIUtil {
    public static ISite getClient(Site site) {
        switch (site) {
            case NFMOVIES:
                return NfMovies.getInstance();
            case TUANZHANG:
                return Tuanzhang.getInstance();
            case DDRK:
                return DDRK.getInstance();
            case DUBOKU:
                return Duboku.getInstance();
        }

        return null;
    }

    public static ISite getClient(String name) {
        switch (name) {
            case NfMovies.NAME:
                return NfMovies.getInstance();
            case Tuanzhang.NAME:
                return Tuanzhang.getInstance();
            case DDRK.NAME:
                return DDRK.getInstance();
            case Duboku.NAME:
                return Duboku.getInstance();
        }
        return null;
    }
}

File: app/src/main/java/com/futuretech/nfmovies/Utils/APIUtil.kt
package com.futuretech.nfmovies.Utils

import com.futuretech.nfmovies.API.*

object APIUtil {
    fun getClient(site: Site): ISite? {
        return when (site) {
            Site.NFMOVIES -> NfMovies.instance
            Site.TUANZHANG -> Tuanzhang.instance
            Site.DDRK -> DDRK.instance
            Site.DUBOKU -> Duboku.instance
        }
    }

    fun getClient(name: String): ISite? {
        when (name) {
            NfMovies.NAME -> return NfMovies.instance
            Tuanzhang.NAME -> return Tuanzhang.instance
            DDRK.NAME -> return DDRK.instance
            Duboku.NAME -> return Duboku.instance
        }
        return null
    }
}
